### PR TITLE
DT-2745: amended update role so that single element of role can be passed in to be updated.

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/RoleResource.kt [hunk 1]
```diff
@@ -36,12 +36,12 @@ class RoleResource(
   private val roleService: RoleService
 ) {
 
-  @PreAuthorize("hasRole('ROLE_MAINTAIN_ACCESS_ROLES_ADMIN')")
+  @PreAuthorize("hasAnyRole('ROLES_ADMIN','ROLE_MAINTAIN_ACCESS_ROLES_ADMIN')")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/RoleResource.kt:39]: this hasn't got a `ROLE_` prefix?

Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/RoleResource.kt:39]: i.e. should it be `ROLE_ROLES_ADMIN`?

**Review by brightonsbox — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/RoleResource.kt (hunk 1 above)
Comment (brightonsbox) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/RoleResource.kt:39]: The integration test is still passing so might

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**